Can't run integration tests of the application using redis

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

<!-- Explain the bug, if an error is being thrown a stack trace helps -->
Can't run integration tests of the application using Redis. The application fails to start because it can't connect to the Redis.
```Java
Caused by: org.redisson.client.RedisConnectionException: Unable to connect to Redis server: localhost/127.0.0.1:6379
	at org.redisson.connection.pool.ConnectionPool$1.lambda$run$0(ConnectionPool.java:159)
	at org.redisson.misc.RedissonPromise.lambda$onComplete$0(RedissonPromise.java:183)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551)
```
##### **Motivation for or Use Case**

<!-- Explain why this is a bug for you -->
All integration tests fail. I can't move forward to develop.

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->
Create a monolithic application that uses Redis as cache. Run integration tests. Test fails because of
```java 
Caused by: org.redisson.client.RedisConnectionException: Unable to connect to Redis server: localhost/127.0.0.1:6379
```

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

Current Jhipster uses Testcontainers to startup a Redis server.  
Both the startup and @DynamicPropertySource code are in a Junit Extension.   
However, the @DynamicPropertySource method can't be recognized by Spring, because it must be in the test class.


##### **JHipster Version(s)**

<!--
Which version of JHipster are you using, is it a regression?
-->
```
jhipster  --version
INFO! Using JHipster version installed globally
6.9.1
```
##### **JHipster configuration**

<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
---
If you have a JDL please wrap it in below structure
  <details>
  <summary>JDL definitions</summary>
  <pre>
     JDL content here
  </pre>
  </details>
-->
```
INFO! Using JHipster version installed locally in current project's node_modules
INFO! Executing jhipster:info
INFO! Options: from-cli: true
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
sam@0.0.1-SNAPSHOT /Users/**/git/template/sam-template
└── generator-jhipster@6.9.1

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.xxx.yyy",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.9.1",
    "applicationType": "monolith",
    "baseName": "sam",
    "packageName": "com.xxx.yyy",
    "packageFolder": "com/xxx/yyy",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "redis",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Memory",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "react",
    "clientTheme": "cerulean",
    "clientThemeVariant": "primary",
    "creationTimestamp": 1593313640874,
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en"],
    "blueprints": [],
    "appsFolders": ["sam-template"],
    "directoryPath": "../",
    "dockerRepositoryName": "",
    "dockerPushCommand": "docker push",
    "kubernetesNamespace": "default",
    "kubernetesServiceType": "Ingress",
    "ingressType": "nginx",
    "ingressDomain": "192.168.99.100.nip.io",
    "monitoring": "prometheus",
    "istio": false
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>


##### **Environment and Tools**

openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.7+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)

git version 2.17.0

node: v12.18.1

npm: 6.14.5

yeoman: 3.1.0

yarn: 1.6.0

Docker version 19.03.8, build afacb8b

docker-compose version 1.25.5, build 8a1c60f6

INFO! Congratulations, JHipster execution is complete!
```
##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->
N/A
##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->
No OS related issue
-   [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/server/templates/src/test/java/package/RedisTestContainerExtension.java.ejs
+++ b/generators/server/templates/src/test/java/package/RedisTestContainerExtension.java.ejs
@@ -16,13 +16,10 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
-
 package <%= packageName %>;
 
 import org.junit.jupiter.api.extension.BeforeAllCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
-import org.springframework.test.context.DynamicPropertyRegistry;
-import org.springframework.test.context.DynamicPropertySource;
 import org.testcontainers.containers.GenericContainer;
 
 import java.util.concurrent.atomic.AtomicBoolean;
@@ -31,18 +28,16 @@ public class RedisTestContainerExtension implements BeforeAllCallback {
 
     private static AtomicBoolean started = new AtomicBoolean(false);
 
-    private static GenericContainer redis;
-
-    @DynamicPropertySource
-    static void redisProperties(DynamicPropertyRegistry registry) {
-        registry.add("jhipster.cache.redis.server", () -> "redis://" + redis.getContainerIpAddress() + ":" + redis.getMappedPort(6379));
-    }
+    private static GenericContainer redis = new GenericContainer("<%= DOCKER_REDIS %>").withExposedPorts(6379);
 
     @Override
     public void beforeAll(ExtensionContext extensionContext) throws Exception {
         if (!started.get()) {
-            redis = new GenericContainer("<%= DOCKER_REDIS %>").withExposedPorts(6379);
             redis.start();
+            System.setProperty(
+                "jhipster.cache.redis.server",
+                "redis://" + redis.getContainerIpAddress() + ":" + redis.getMappedPort(6379)
+            );
             started.set(true);
         }
     }

